metagraph/src/cli/align.cpp

```diff
@@ -211,15 +211,12 @@ void map_sequences_in_file(const std::string &file,
 
         if (config.count_kmers) {
             std::sort(graphindices.begin(), graphindices.end());
-            size_t num_unique_matching_kmers = std::inner_product(
-                graphindices.begin() + 1, graphindices.end(),
-                graphindices.begin(),
-                size_t(graphindices.front() != DeBruijnGraph::npos),
-                std::plus<size_t>(),
-                [](DeBruijnGraph::node_index next, DeBruijnGraph::node_index prev) {
-                    return next != DeBruijnGraph::npos && next != prev;
-                }
-            );
+            size_t num_unique_matching_kmers = 0;
+            auto prev = DeBruijnGraph::npos;
+            for (auto i : graphindices) {
+                if (i != DeBruijnGraph::npos && i != prev)
+                    ++num_unique_matching_kmers;
+            }
```

Comment:
hold on, `prev` is never updated. How does this pass tests?
```suggestion
            auto prev = DeBruijnGraph::npos;
            for (auto i : graphindices) {
                if (i != DeBruijnGraph::npos && i != prev)
                    ++num_unique_matching_kmers;
                prev = i;
            }
```

Reply:
ah, I think there's no integration test for this

Reply:
Could you add one or two?

Maybe first before fixing this error, to make sure they would fail.

Reply:
ok, added